coredns/plugin/zone_test.go

```diff
@@ -0,0 +1,222 @@
+				"aaaa.example.com.\t60\tIN\tAAAA\t2001:db8::68": {},
+			},
+		},
+		{
+			name: "insert TXT record",
+			args: args{
+				ep: &endpoint.Endpoint{
+					DNSName:    "txt.example.com",
+					Targets:    []string{"foo=bar"},
+					RecordType: endpoint.RecordTypeTXT,
+					RecordTTL:  60,
+				},
+			},
+			expectedRRData: map[string]rrData{
+				"txt.example.com.\t60\tIN\tTXT\t\"foo=bar\"": {},
+			},
+		},
+		{
+			name: "insert A record with multiple targets",
+			args: args{
+				ep: &endpoint.Endpoint{
+					DNSName:    "a.example.com",
+					Targets:    []string{"1.1.1.1", "2.2.2.2"},
+					RecordType: endpoint.RecordTypeA,
+					RecordTTL:  60,
+				},
+			},
+			expectedRRData: map[string]rrData{
+				"a.example.com.\t60\tIN\tA\t1.1.1.1": {},
+				"a.example.com.\t60\tIN\tA\t2.2.2.2": {},
+			},
+		},
+		{
+			name: "insert A record with geo",
+			args: args{
+				ep: &endpoint.Endpoint{
+					DNSName:    "a.example.com",
+					Targets:    []string{"1.1.1.1"},
+					RecordType: endpoint.RecordTypeA,
+					RecordTTL:  60,
+					ProviderSpecific: endpoint.ProviderSpecific{
+						{
+							Name:  "geo-code",
+							Value: "GEO-EU",
+						},
+					},
+				},
+			},
+			expectedRRData: map[string]rrData{
+				"a.example.com.\t60\tIN\tA\t1.1.1.1": {
+					geo: ptr.To("GEO-EU"),
+				},
+			},
+		},
+		{
+			name: "insert A record with weight",
+			args: args{
+				ep: &endpoint.Endpoint{
+					DNSName:    "a.example.com",
+					Targets:    []string{"1.1.1.1"},
+					RecordType: endpoint.RecordTypeA,
+					RecordTTL:  60,
+					ProviderSpecific: endpoint.ProviderSpecific{
+						{
+							Name:  "weight",
+							Value: "100",
+						},
+					},
+				},
+			},
+			expectedRRData: map[string]rrData{
+				"a.example.com.\t60\tIN\tA\t1.1.1.1": {
+					weight: ptr.To(int64(100)),
+				},
+			},
+		},
+		{
+			name: "insert A record with weight and geo (prefer weight)",
+			args: args{
+				ep: &endpoint.Endpoint{
+					DNSName:    "a.example.com",
+					Targets:    []string{"1.1.1.1"},
+					RecordType: endpoint.RecordTypeA,
+					RecordTTL:  60,
+					ProviderSpecific: endpoint.ProviderSpecific{
+						{
+							Name:  "weight",
+							Value: "100",
+						},
+						{
+							Name:  "geo-code",
+							Value: "GEO-EU",
+						},
+					},
+				},
+			},
+			expectedRRData: map[string]rrData{
+				"a.example.com.\t60\tIN\tA\t1.1.1.1": {
+					weight: ptr.To(int64(100)),
+				},
+			},
+		},
+		{
+			name: "insert CNAME record with geo",
+			args: args{
+				ep: &endpoint.Endpoint{
+					DNSName:    "cname.example.com",
+					Targets:    []string{"t.example.com"},
+					RecordType: endpoint.RecordTypeCNAME,
+					RecordTTL:  60,
+					ProviderSpecific: endpoint.ProviderSpecific{
+						{
+							Name:  "geo-code",
+							Value: "GEO-EU",
+						},
+					},
+				},
+			},
+			expectedRRData: map[string]rrData{
+				"cname.example.com.\t60\tIN\tCNAME\tt.example.com.": {
+					geo: ptr.To("GEO-EU"),
+				},
+			},
```

Comment:
this is a good set of tests. Should we include inserting a duplicate?

Reply:
Yeas i have some other tests coming, will add that as part of the next set of changes.